typescript/lib/parsers/hf.ts

```diff
@@ -0,0 +1,227 @@
+import {
+  HfInference,
+  TextGenerationArgs,
+  TextGenerationOutput,
+  TextGenerationStreamOutput,
+} from "@huggingface/inference";
+
+import {
+  Prompt,
+  Output,
+  PromptInput,
+  ModelMetadata,
+  ExecuteResult,
+} from "../../types";
+import { CompletionCreateParams } from "openai/resources";
+import _ from "lodash";
+import { getAPIKeyFromEnv } from "../utils";
+import { ParameterizedModelParser } from "../parameterizedModelParser";
+import { JSONObject } from "../../common";
+import { AIConfigRuntime } from "../config";
+import { InferenceOptions } from "../modelParser";
+
+export class HuggingFaceTextGenerationModelParser extends ParameterizedModelParser<TextGenerationArgs> {
+  private hfClient: HfInference;
+
+  public constructor(modelId: string, use_api_token?: Boolean) {
+    super();
+    let token = use_api_token
+      ? getAPIKeyFromEnv("HUGGING_FACE_API_TOKEN")
+      : undefined;
+    this.hfClient = new HfInference(token);
```

Comment:
Should we instantiate this in `run`? So that the API Key isn't needed for serialize and deserialize?

Reply:
Yes that would make sense. Can instantiate on first run by checking if client isn't defined, and if not define client.